src/transformers/modeling_tf_utils.py

```diff
@@ -554,9 +559,226 @@ def input_processing(func, config, input_ids, **kwargs):
+        model_layer_map (`Dict`): A dictionnary mapping the layer name to the index of the layer in the model.
+        resolved_archive_file (`str`): Path to the checkpoint file from which the weights will be loaded
+        ignore_mismatched_sizes (bool, optional):
+            _description_. Defaults to False. Whether to ignore the mismatch keys
+
+    Returns:
+        Three lists, one for the layers that were found and succesfully restored (from the shard file), one for the
+        missmatched layers, and another one for the unexpected layers.
+    """
+    saved_weight_names_set = set()
+    saved_weights = {}
+    missmatched_keys = set()
+    unexpected_keys = set()
+    # Read the H5 file
+    with h5py.File(resolved_archive_file, "r") as f:
+        # Retrieve the name of each layer from the H5 file
+        saved_h5_model_layers_name = set(hdf5_format.load_attributes_from_hdf5_group(f, "layer_names"))
+        weight_value_tuples = []
+
+        # Compute missing and unexpected sub layers
+        # Store the weights in list of tuples that looks like [(weight_object, value_of_weight),...]
+        for layer_name in saved_h5_model_layers_name:
+            h5_layer_object = f[layer_name]
+            saved_weights[layer_name] = np.asarray(h5_layer_object)
+
+            saved_weight_names_set.add(layer_name)
+
+            if layer_name not in model_layer_map:
+                unexpected_keys.add(layer_name)
+            else:
+                symbolic_weight = model.weights[model_layer_map[layer_name]]
+
+                saved_weight_value = saved_weights[layer_name]
+                # If the current weight is found
+                if saved_weight_value is not None:
+                    # Check if the shape of the current weight and the one from the H5 file are different
+                    if K.int_shape(symbolic_weight) != saved_weight_value.shape:
+                        # If yes we reshape the weight from the H5 file accordingly to the current weight
+                        # If the two shapes are not compatible we raise an issue
+                        try:
+                            array = np.reshape(saved_weight_value, K.int_shape(symbolic_weight))
+                        except ValueError as e:
+                            if ignore_mismatched_sizes:
+                                missmatched_keys.add(
+                                    (layer_name, saved_weight_value.shape, K.int_shape(symbolic_weight))
+                                )
+                                continue
+                            else:
+                                raise e
+                    else:
+                        array = saved_weight_value
+
+                # We create the tuple that will be loaded and add it to the final list
+                weight_value_tuples.append((symbolic_weight, array))
+
+    K.batch_set_value(weight_value_tuples)
+
+    return saved_weight_names_set, unexpected_keys, missmatched_keys
```

Comment:
There are some exceptions in the PT counterpart that would probably be cool to have here! 👉  [PT exceptions](https://github.com/huggingface/transformers/blob/d3cb28886ac68beba9a6646b422a4d727b056c0c/src/transformers/modeling_utils.py#L462)

Reply:
Good point ;)